modules/pool-cidr/main.tf

```diff
@@ -7,3 +7,29 @@ resource "aws_vpc_ipam_pool_cidr" "this" {
   ipam_pool_id   = var.ipam_pool_id
   netmask_length = var.netmask_length
 }
+
+data "aws_vpc_ipam_pool_cidrs" "this" {
+  ipam_pool_id = var.ipam_pool_id
+  filter {
+    name   = "cidr"
+    values = [var.cidr_filter_pattern]
+  }
+}
+
+locals {
+  mycidrs = [for cidr in data.aws_vpc_ipam_pool_cidrs.this.ipam_pool_cidrs : cidr.cidr if cidr.state == "provisioned"]
+}
+
+resource "aws_ec2_managed_prefix_list" "this" {
```

Comment:
What is the concept for this resource? Not sure I understand the use case being filled here, in the context of IPAM.